File: software/examples/java/ExampleSimple.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.TNGDI8;

public class ExampleSimple {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your TNG DI8
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		TNGDI8 di8 = new TNGDI8(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get current values
		boolean[] values = di8.getValues(); // Can throw com.tinkerforge.TimeoutException

		System.out.println("Channel 0: " + values[0]);
		System.out.println("Channel 1: " + values[1]);
		System.out.println("Channel 2: " + values[2]);
		System.out.println("Channel 3: " + values[3]);
		System.out.println("Channel 4: " + values[4]);
		System.out.println("Channel 5: " + values[5]);
		System.out.println("Channel 6: " + values[6]);
		System.out.println("Channel 7: " + values[7]);

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}

File: software/examples/javascript/ExampleSimple.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your TNG DI8

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var di8 = new Tinkerforge.TNGDI8(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Get current values
        di8.getValues(
            function (values) {
                console.log('Channel 0: ' + values[0]);
                console.log('Channel 1: ' + values[1]);
                console.log('Channel 2: ' + values[2]);
                console.log('Channel 3: ' + values[3]);
                console.log('Channel 4: ' + values[4]);
                console.log('Channel 5: ' + values[5]);
                console.log('Channel 6: ' + values[6]);
                console.log('Channel 7: ' + values[7]);
            },
            function (error) {
                console.log('Error: ' + error);
            }
        );
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
